Platform/Interfaces/iadc.hpp

```diff
@@ -0,0 +1,44 @@
+/*
+ * STM32 F4 ADC ABSTRACT INTERFACE
+ * Author:   William Kim
+ *
+ * Email:    wkim@utdallas.edu
+ *
+ * (c) 2024 Dallas Formula Racing - Embedded Firmware Team
+ * Formula SAE International Collegiate Chapter
+ * GPL-3.0 License
+ */
+
+#ifndef IADC_H
+#define IADC_H
+
+#include <cstdint>
+#include <memory>
+#include <vector>
+
+namespace platform
+{
+
+    class IADC
+    {
+    public:
+        virtual ~IADC() = 0;
+
+        //adc operations
+        virtual uint16_t convert() const = 0;
+        virtual void writeToBuffer() = 0;
+
+    private:
+
+    protected:
+    std::shared_ptr<std::vector<uint16_t>> adc_buf; //container for adc read
```

Comment:
I'm wondering why the adc_buffer would be a shared pointer?

Reply:
made shared pointer so different threats could use it at once. also because shared pointers avoid deep copies when referenced by multiple other functions.